core/chains/evm/logpoller/log_poller_test.go

```diff
@@ -1557,9 +1562,13 @@ func TestTooManyLogResults(t *testing.T) {
 		Message: "query returned more than 10000 results. Try with this block range [0x100E698, 0x100E6D4].",
 	}
 
+	// Simulate currentBlock = 300
+	head := &evmtypes.Head{Number: 300}
+	finalized := &evmtypes.Head{Number: head.Number - lpOpts.FinalityDepth}
+	headTracker.On("LatestAndFinalizedBlock", mock.Anything).Return(head, finalized, nil).Once()
 	call1 := ec.On("HeadByNumber", mock.Anything, mock.Anything).Return(func(ctx context.Context, blockNumber *big.Int) (*evmtypes.Head, error) {
 		if blockNumber == nil {
-			return &evmtypes.Head{Number: 300}, nil // Simulate currentBlock = 300
+			panic("unexpected call to get current head")
```

Comment:
We shouldn't call `panic` in tests, because it will cause all other tests running in parallel to halt prematurely... potentially with incomplete or corrupt logs.

I think `require.FailNowf("unexpected call to get current head")` is the right way to do this